lib/xenops_server.ml

```diff
@@ -804,7 +804,14 @@ let simplify f =
 
 let rec atomics_of_operation = function
 	| VM_start id ->
-		[
+		(
+			let module B = (val get_backend () : S) in
+			let power = (B.VM.get_state (VM_DB.read_exn id)).Vm.power_state in
+			begin match power with
+				| Running -> raise (Bad_power_state(Running, Halted))
```

Comment:
I don't see `atomics_of_operation` raise otherwise an exception and this makes me wonder whether this is the right way to handle this. I think @robhoes should take a look here.